esp-hal/src/dma/pdma/crypto.rs

```diff
@@ -0,0 +1,525 @@
+use esp32s2::crypto_dma::aes_sha_select::SELECT;
+use portable_atomic::{AtomicBool, Ordering};
+
+use crate::{
+    asynch::AtomicWaker,
+    dma::*,
+    interrupt::Priority,
+    peripheral::Peripheral,
+    peripherals::Interrupt,
+};
+
+pub(super) type CryptoRegisterBlock = crate::peripherals::crypto_dma::RegisterBlock;
+
+/// The RX half of a Crypto DMA channel.
+pub struct CryptoDmaRxChannel(pub(crate) CryptoDmaChannel);
+
+impl crate::private::Sealed for CryptoDmaRxChannel {}
+impl DmaRxChannel for CryptoDmaRxChannel {}
+impl Peripheral for CryptoDmaRxChannel {
+    type P = Self;
+
+    unsafe fn clone_unchecked(&self) -> Self::P {
+        Self(self.0.clone_unchecked())
+    }
+}
+
+/// The TX half of a Crypto DMA channel.
+pub struct CryptoDmaTxChannel(pub(crate) CryptoDmaChannel);
+
+impl crate::private::Sealed for CryptoDmaTxChannel {}
+impl DmaTxChannel for CryptoDmaTxChannel {}
+impl Peripheral for CryptoDmaTxChannel {
+    type P = Self;
+
+    unsafe fn clone_unchecked(&self) -> Self::P {
+        Self(self.0.clone_unchecked())
+    }
+}
+
+impl RegisterAccess for CryptoDmaTxChannel {
+    fn reset(&self) {
+        let register_block = self.0.register_block();
+        register_block.conf().modify(|_, w| {
+            w.out_rst().set_bit();
+            w.ahbm_rst().set_bit();
+            w.ahbm_fifo_rst().set_bit()
+        });
+        register_block.conf().modify(|_, w| {
+            w.out_rst().clear_bit();
+            w.ahbm_rst().clear_bit();
+            w.ahbm_fifo_rst().clear_bit()
+        });
+    }
+
+    fn set_burst_mode(&self, burst_mode: BurstConfig) {
+        let register_block = self.0.register_block();
+        register_block
+            .conf()
+            .modify(|_, w| w.out_data_burst_en().bit(burst_mode.is_burst_enabled()));
+    }
+
+    fn set_descr_burst_mode(&self, burst_mode: bool) {
+        let register_block = self.0.register_block();
+        register_block
+            .conf()
+            .modify(|_, w| w.outdscr_burst_en().bit(burst_mode));
+    }
+
+    fn set_peripheral(&self, peripheral: u8) {
+        let peripheral = match peripheral {
+            p if p == DmaPeripheral::Aes as u8 => SELECT::Aes,
+            p if p == DmaPeripheral::Sha as u8 => SELECT::Sha,
+            _ => unreachable!(),
+        };
+        let register_block = self.0.register_block();
+        register_block
+            .aes_sha_select()
+            .modify(|_, w| w.select().variant(peripheral));
```

Comment:
We should consider doing something like this for ESP32's SPI_DMA

Reply:
We'll do that once it matters, but it's not necessary to waste CPU cycles to allow mixing DMA channels right now.